### ISSUE TITLE
The layout of the bar is broken if use custom view for *BarButtonItems

### ISSUE THREAD

**Issue by YuriyVelichkoPI** [LeoNatan/LNPopupController#189]
#### Description

Description in the title.
The bug appeared in the build for iOS 11 with Xcode 9. In previous builds all worked fine - all views are placed in proper positions.
LNPopupController version - 2.5.0 (integrated through Carthage)

#### Steps to Reproduce
In the example project, in the method _setPopupItemButtonsWithTraitCollection, need to set a custom view for button item:

`self.popupItem.rightBarButtonItems = @[[[UIBarButtonItem alloc] initWithCustomView:[[UIView alloc] initWithFrame:CGRectMake(0, 0, 20, 20)]]];`

Result:
![screen shot 2017-09-26 at 15 16 14](https://user-images.githubusercontent.com/21169695/30860141-c1e10a16-a2ce-11e7-8ea0-23f1c5b096cb.png)

#### Source of error and partial solution

Tthe problem description and the partial solution can be found here:
https://stackoverflow.com/questions/46247451/ios-11-uibarbuttonitem-images-not-sizing

But it is not enough. If to set constraints like this:

    UIView* view = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 20, 20)];
    if (@available(iOS 9, *)) {
        [view.widthAnchor constraintEqualToConstant: 20].active = YES;
        [view.heightAnchor constraintEqualToConstant: 20].active = YES;
    }
    self.popupItem.rightBarButtonItems = @[[[UIBarButtonItem alloc] initWithCustomView:view]];

then, the result will look better:

![screen shot 2017-09-26 at 15 30 10](https://user-images.githubusercontent.com/21169695/30860457-d48cf9e4-a2cf-11e7-9e43-2cc2962cb15b.png)

**But still incorect**:

![screen shot 2017-09-26 at 15 32 38](https://user-images.githubusercontent.com/21169695/30860508-f7df6ab2-a2cf-11e7-96f4-96287e0b2627.png)

The width of the text labels is too small. So that, the text is not aligned to the center of the bar.

Looks like the error lies in method `_layoutTitles` on calculating the value of `_titlesView.frame` for iOS 11.

**Comment by LeoNatan** [comment]
Since frame should be correct even with auto layout, I don’t see a reason for a bug. I will look at this soon.

**Comment by wfratczak** [comment]
I faced the same issue after update to iOS 11 iPhone 6s/7. For iPhone 5s works well.

Screenshot:
<img width="645" alt="screen shot 2017-09-28 at 9 26 11 am" src="https://user-images.githubusercontent.com/9217905/30969187-a0c6f73e-a42f-11e7-9e02-2d6873e7a088.png">

View debugger:
<img width="398" alt="screen shot 2017-09-28 at 9 24 51 am" src="https://user-images.githubusercontent.com/9217905/30969188-a0d258d6-a42f-11e7-967f-8cb8d8e838e6.png">

**Comment by LeoNatan** [comment]
Can you please attach a demo project?

**Comment by wfratczak** [comment]
I can't attach my project, the issue appears when I attach custom rightBarButtonItem. 
When I add widthConstraint for this rightItem, everything works well.

**Comment by LeoNatan** [comment]
This is an iOS 11 requirement as I understand it. So not sure it’s a bug.

**Comment by YuriyVelichkoPI** [comment]
@LeoNatan from my opinion, the bug is that the text label(s) is not centered in the bar if to add a custom view. Even with applying iOS11 requirements (with using constraints instead of the frame). 
The position and width of labels are calculated in the code of framework. So looks like calculation can be incorrect in case of new iOS approach of managing bar button items. 

Or, can you suggest another path to support iOS11 and custom views? It will be great!

As I mentioned you can reproduce the issue easily with example project from the current repository.

**Comment by MossDevlper** [comment]
Hey guys, I´m still not sure what causes this weird behavior, but it is true it has to do with title/subtitle frame. I want to share with you guys how i manages to fix it, while waiting for an oficial solution from @LeoNatan. Hope its useful for someone else. 

In the class **LNPopupBar.m** I edited like in the screenshot.

![screen shot 2017-11-08 at 5 52 29 pm](https://user-images.githubusercontent.com/13543812/32581105-1927f216-c4ae-11e7-91cf-685d02e30adb.png)

**Comment by LeoNatan** [comment]
Thanks for the continued discussion, guys.
Could you guys please provide a demo project where these issues reproduce? Just want to see how you create your bar button items.

**Comment by MossDevlper** [comment]
Unfortunately I don´t have a demo at hand, but I will gladly show you the way I create the **UIBarButtonItem**:

Inside my RootViewController I use these commando to open the PopupBar.

![screen shot 2017-11-08 at 8 21 39 pm](https://user-images.githubusercontent.com/13543812/32585324-8f0be6ea-c4c2-11e7-8c3d-e5bbcf56e97d.png)

Inside my **_playerVC_** I then create the views I´ll insert the UIBarButtonItem.

![screen shot 2017-11-08 at 8 22 01 pm](https://user-images.githubusercontent.com/13543812/32585347-a987ffa4-c4c2-11e7-9d0b-6e33a3ae839b.png)

And for last I make a call to my **_initToolbarVariables_**, where I insert and set my popupItem.

![screen shot 2017-11-08 at 8 22 15 pm](https://user-images.githubusercontent.com/13543812/32585388-d8973f30-c4c2-11e7-86f7-1c6d5c75c68b.png)

Hope this helps you.

**Comment by RiccardoManoni** [comment]
I got the same issue since I updated Xcode to version 9.x and iOS to version 11.x. I'm using LNPopupController v1.4.6 (integrated through Carthage) but to me the problem appears only when i need to update an UIBarButtonItem, for example when I want to change playButton with pauseButton like this:

(from)
popupItem.leftBarButtonItems = [playButton, nextButton]

(to)
popupItem.leftBarButtonItems.removeAll()
popupItem.leftBarButtonItems = [pauseButton, nextButton]

I discovered that the issue disappears if instead of use removeAll() I simply update the specific item that I want to change, for example:

(from)
popupItem.leftBarButtonItems = [playButton, nextButton]

(to)
popupItem.leftBarButtonItems![0] = pauseButton

That way everything works good like before.
Hope this helps you.
